### PR TITLE
HPCC-25342 Capture eclcc logging in containerized mode

### PR DESCRIPTION
**Pull request by jakesmith**
In containerized mode, turn off file creation and
association of file into workunit.
[ logging will still be output and seen by K8s ]

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
change looks good, but should this be master rather than 7.12.x?  @richardkchapman?

**Comment by richardkchapman** [comment]
I'd be happier with master. Not 100% sure about the change even there

**Review by jakesmith — COMMENTED** [review]
@ghalliday - retargeted against master

File: ecl/eclccserver/eclccserver.cpp [hunk 1]
```diff
@@ -339,8 +339,17 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
         eclccCmd.append(" --timings --xml");
         eclccCmd.append(" --nostdinc");
         eclccCmd.append(" --metacache=");
-        VStringBuffer logfile("%s.eclcc.log", workunit->queryWuid());
+
+#ifdef _CONTAINERIZED
+        eclccCmd.append(" --nologfile");
```
Comment (ghalliday) [inline ecl/eclccserver/eclccserver.cpp:344]: Have you checked the output?  the output to stdout is not in the standard logging format - e.g., it doesn't include the id/audience etc.  That should be addressed if the output from eclcc is going to be sent to the standard log processing.

**Review by ghalliday — COMMENTED** [review]
@jakesmith see comment

File: ecl/eclccserver/eclccserver.cpp (hunk 1 above)
Comment (jakesmith) [reply ecl/eclccserver/eclccserver.cpp]: @ghalliday - please see new commit.

I have checked in bare-metal / containerized.
In bare-metal logging is captured to a file as before, but moved to /var/log/.. (prev. in /var/lib).
In containerized, logging is output to stdout and seen by K8s.

File: ecl/eclcc/eclcc.cpp [hunk 2]
```diff
@@ -673,21 +674,33 @@ void EclCC::loadOptions()
...
-            if (logVerbose)
-                fprintf(stdout, "Logging to '%s'\n",lf.str());
+            if (optLogfile.length())
+            {
+                StringBuffer lf;
+                openLogFile(lf, optLogfile, optLogDetail, false);
+                if (logVerbose)
+                    fprintf(stdout, "Logging to '%s'\n",lf.str());
+            }
+            if (optMonitorInterval)
```
Comment (ghalliday) [inline ecl/eclcc/eclcc.cpp:699]: Should this still be happening in containerized?  Probably useful until the metrics are fully functional.

**Review by ghalliday — COMMENTED** [review]
@jakesmith a couple of minor comments.  Worth squashing at the same time.

File: ecl/eclcc/eclcc.cpp (hunk 2 above)
Comment (jakesmith) [reply ecl/eclcc/eclcc.cpp:699]: I'm not sure useful, or will be replaced by metrics, there may well be other K8s metrics for free (dashboard) that supplant most of what it does already.
But should prob revisit all places that issue such logging separately later when metrics has progressed.

File: ecl/eclcc/eclcc.cpp [hunk 3]
```diff
@@ -2708,6 +2721,9 @@ int EclCC::parseCommandLineOptions(int argc, const char* argv[])
         else if (iter.matchFlag(optNoLogFile, "--nologfile"))
         {
         }
+        else if (iter.matchFlag(optLogToStdOut, "--logToStdOut"))
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply ecl/eclcc/eclcc.cpp:2724]: will change

**Comment by jakesmith** [comment]
@ghalliday - changed logToStdOut to lowercase, and reverted bare-metal location of log to /var/log/..
Please review

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash.

**Comment by jakesmith** [comment]
@richardkchapman - squashed ready to merge

**Pull request merged**